pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -157,6 +158,35 @@ func (r *ReconcileAWSFederatedAccountAccess) Reconcile(request reconcile.Request
 		return reconcile.Result{}, nil
 	}
 
+	// Check if the FAA has the uuid label
+	if !hasUUIDLabel(currentFAA) {
+		// Generate a new UUID
+		uuid := rand.String(6)
+
+		reqLogger.Info(fmt.Sprintf("Adding UUID %s to AccountAcess %s", uuid, currentFAA.Name))
+		newLabel := map[string]string{"uuid": uuid}
+
+		// Join the new UUID label with any current labels
+		if currentFAA.Labels != nil {
+			currentFAA.Labels = joinLabelMaps(currentFAA.Labels, newLabel)
```

Comment:
This could be cleaned up by putting the check login in the joinLabelMaps function. note for future improvements.